.github/composites/docker-build-push/action.yaml

```diff
@@ -0,0 +1,131 @@
+name: 'Build and Publish Docker Image'
+description: 'Build and Publish Docker Image'
+inputs:
+  CUSTODIAN_IMAGE_VAR:
+    description: 'env var name for custodian image'
+    required: true
+  dockerfile:
+    description: 'Dockerfile to build'
+    required: true
+  repository:
+    description: 'docker hub repository'
+    required: true
+  image:
+    description: 'image'
+    required: true
+  name:
+    description: 'image name'
+    required: true
+  description:
+    description: 'image description'
+    required: true
+  push:
+    description: 'true/false'
+    required: true
+    default: false
+runs:
+  using: "composite"
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Install Docker Build Env
+      # bin directory is in .dockerignore
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip click
+        mkdir -p bin
+        wget -q -O bin/trivy.tgz https://github.com/aquasecurity/trivy/releases/download/v0.5.4/trivy_0.5.4_Linux-64bit.tar.gz
+        cd bin && tar xzf trivy.tgz
+
+    - name: Login to Docker Hub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
+      with:
+        username: ${{ env.HUB_USER }}
+        password: ${{ env.HUB_TOKEN }}
+
+    - name: determine tags
+      shell: bash
+      id: calctags
+      run: |
+        python tools/dev/dockerpkg.py tags -t auto
+        echo "::set-output name=tags::$(python tools/dev/dockerpkg.py tags -t auto)"
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        # list of Docker images to use as base name for tags
+        images: |
+          ${{ inputs.repository }}/${{ inputs.image }}
+        labels: |
+          org.opencontainers.image.documentation=https://cloudcustodian.io/docs
+          # we overwrite the default labels here, defaults are description:github description and title:"cloud-custodian"
+          org.opencontainers.image.description=${{ inputs.description }}
```

Comment:
can we make this image metadata (name, description) static in the docker files instead of passing it through in builds.